test/metric/metric_value_query.go

```diff
@@ -11,23 +11,29 @@ import (
 	"log"
 	"time"
 
+	"github.com/aws/amazon-cloudwatch-agent-test/environment"
+	"github.com/aws/amazon-cloudwatch-agent-test/test"
 	"github.com/aws/aws-sdk-go-v2/aws"
 	"github.com/aws/aws-sdk-go-v2/service/cloudwatch"
 	"github.com/aws/aws-sdk-go-v2/service/cloudwatch/types"
-
-	"github.com/aws/amazon-cloudwatch-agent-test/test"
 )
 
 var metricValueFetchers = []MetricValueFetcher{
 	&CPUMetricValueFetcher{},
 	&MemMetricValueFetcher{},
 	&ProcStatMetricValueFetcher{},
 	&DiskIOMetricValueFetcher{},
+	&ContainerInsightsValueFetcher{},
+}
+
+type MetricFetcherFactory struct {
+	Env *environment.MetaData
 }
 
-func GetMetricFetcher(metricName string) (MetricValueFetcher, error) {
+func (f *MetricFetcherFactory) GetMetricFetcher(metricName string) (MetricValueFetcher, error) {
```

Comment:
nit - I think it's confusing to use the same variable name for methods on different structs. `f` is used for `MetricFetchFactory` and `baseMetricValueFetcher` in the same file

Reply:
I can change that [TODO]